api/api/src/main/scala/uk/ac/wellcome/platform/api/services/ElasticsearchService.scala

```diff
@@ -69,6 +70,35 @@ class ElasticsearchService(elasticClient: ElasticClient)(
         }
     }
 
+  def executeMultiSearchRequest(
+    request: MultiSearchRequest): Future[Either[ElasticError, List[SearchResponse]]] =
+    spanFuture(
+      name = "ElasticSearch#executeQuery",
+      spanType = "request",
+      subType = "elastic",
+      action = "query"
+    ) {
+      debug(s"Sending ES multirequest: ${request.show}")
+      val transaction = Tracing.currentTransaction
+      withActiveTrace(elasticClient.execute(request))
+        .map(toEither)
+        .map { response =>
+          response.right.flatMap { case MultiSearchResponse(items) =>
+             val results = items.map(_.response)
+             val error = results.collectFirst { case Left(err) => err }
+             error match {
+               case Some(err) => Left(err)
+               case None => Right(
+                 results.collect { case Right(resp) =>
+                   transaction.addLabel("elasticTook", resp.took)
```

Comment:
@jamieparkinson any thoughts on how we want tracing to work with [ES multi search API](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-multi-search.html)? An alternative would be to sum these values before adding it to the transaction.

Reply:
Is this `took` per request?
Having them summed would be good, as that analysis in the console will be annoying.

Reply:
Yup so currently there would be 3 separate timings (for the children, ancestors and siblings requests),

Reply:
If we had an ID or something to then sum them, so we could average out on how long this whole request takes, that would be useful as that's what people will be experiencing.